src/runtimes/node.js

```diff
@@ -1,38 +1,92 @@
-const { join, basename } = require('path')
+const { basename, dirname, join } = require('path')
 
-const cpFile = require('cp-file')
+const commonPathPrefix = require('common-path-prefix')
 
+const { bundleJsFile } = require('../bundler')
+const { getDependencies, listFilesUsingLegacyBundler } = require('../node_dependencies')
 const { zipNodeJs } = require('../zip_node')
 
-const zipJsFunction = async function ({
-  srcPath,
+const getSrcFiles = function ({ srcPath, mainFile, srcDir, stat, pluginsModulesPath, useEsbuild, externalModules }) {
+  if (!useEsbuild) {
+    return listFilesUsingLegacyBundler({ srcPath, mainFile, srcDir, stat, pluginsModulesPath })
+  }
+
+  if (externalModules.length !== 0) {
+    return getDependencies(mainFile, srcDir, pluginsModulesPath, externalModules)
```

Comment:
How is this impacting bundle time? Are we running both the old bundling logic and the new one in this case?
Is the purpose of this code to copy external dependencies? Will a copy files operation work?

Reply:
> How is this impacting bundle time?

Every external module will carry a performance penalty, because it needs to fall back to the legacy method of traversing the module's dependency tree using Node. We're now doing for external modules what we were previously doing for every module, so unless we're adding every module as external, we're still improving the current bundle times.

> Are we running both the old bundling logic and the new one in this case?

Not really. Since we're flagging these modules as external, esbuild will simply leave them alone, avoiding the cost of traversing and inlining them. So effectively we're only running the old bundling logic for these modules.

> Is the purpose of this code to copy external dependencies?

Yes.

> Will a copy files operation work?

Yes, but we need to copy _all_ the files, including transitive dependencies, for each of these modules. To do this computation, I figured it would make sense to reuse the logic in `getDependencies`.

Reply:
> Also would using something similar to https://github.com/netlify/cli/blob/8b0e0ad2b3cc4d9eec20554afb99114fd5320865/tests/command.dev.test.js#L94 work for the tests?

Yes, that's a possibility! We have some tests that we only want to run for one of the bundlers, but that's still achievable with that approach. I can revisit this, but I'd probably do it after we iron out the rest of the implementation details, in a separate PR, if that's okay?

Reply:
Sure, doing it in a separate PR works.

Reply:
@erezrokah I've changed the bundling mechanism for external modules, as per our conversation, in https://github.com/netlify/zip-it-and-ship-it/pull/325/commits/bd9cad72500b707bd74272144c4be81c296482d7.